### DISCUSSION TITLE
necessary changes for future of basecamps

### DISCUSSION THREAD

**Discussion by Xaleth** [CleverRaven/Cataclysm-DDA#48975] [Ideas]
I think the basecamp shouldn't strictly follow predesigned blueprints, like the builds should be flexible and show the player's design

**Comment by actual-nh** [comment]
More details? Ideas as to how this could be done (including desired interface)?

**Comment by sashakiller33** [comment]
I can think of the two ways it could be implemented. One is we call some place a base and using building systems, that we have(zone build), we build by our hands and base interface just makes count of different things we have. Like how many beds we have, do we have fire place and etc.
Or add internal base blueprints builder, where you can create and edit json plans of a bases.

**Comment by krulunio** [comment]
I was thinking about changing how basecamps work, but I'm currently lacking time.
My ideas are:
1. Bases and expansions won't change OMT; it's unnecessary
2. All OMTs will be valid base/expansion locations; whether NPCs can work in that OMT or not will depend on presence of hostile mobs and non-friendly NPCs
3. Any OMT could be any expasion; there would be only small number of nonredundant expansions, giving access to specific missions or recipes
4. Each camp could have, in theory, infinite number of expansions; conditions for acquiring expansions would be that new expansion must be adjacent to base/expansion (even across Z-levels) and,maybe as it is currently, 2 beds for each expansion
5. Each camp would spawn bandits and hostile mobs nearby; how much and fast they spawn would depend on number of expansions, amount of food, distance to city, attitude towards other factions (NPCs from hostile nonbandit factions could also attack camp)
6. More stats:
water - works same as calories
health - when below 0, NPCs won't start missions/crafts (so players won't depend only on alcohol and mutant meat to feed companions), also would be used in healing missions
happiness - when below 0, NPCs won't start missions/crafts, I'm not sure how it would exactly work
electricity - gained from nuclear reactors, it would be assigned to specific expansion to gain some effects or stats happiness from specific furniture i.e. arcade machines (when proper electric system is implemented, it will become obsolete, probably)
beds - aside from expansions, beds would be used to calculate how many NPCs can work in camp at the same time
7. New camp menu design; it would be divided into 4 tabs: general, expansions, crafting, missions :
general - shows info about camp, crafts/upgrades/missions in progress and options (abandon camp, distribute food/consumables, rename camp, join camps)
expansions - shows list of expansions and options for manipulating expansions (add, remove, rename expansion)
crafting - shows list of available crafting recipes
missions - shows list of available upgrades/missions
8. Add construction mission; it would send NPC with resources to build things specified by construction zones, allowing construction of anything in basecamp through basecamp
9. Add more missions, like healing NPCs, trading with factions, installing bionics
10. Allow using vehicles/animals to speed up "travel missions" (cutting logs, scouting, patroling, trading with factions)
11. Furniture used by expansion will need to be selected with faction storage zone, to omptimize checking what is available in expansion
12. Bed would be defined as "bed furniture" (straw bed, mattress, sofa, etc.) of same type connected together; seach bed would give some amount of "happiness", depending on size of bed
